pkg/controller/backup-operator/s3_backup.go

```diff
@@ -23,20 +24,34 @@ import (
 	"github.com/coreos/etcd-operator/pkg/backup/backupapi"
 	"github.com/coreos/etcd-operator/pkg/backup/writer"
 	"github.com/coreos/etcd-operator/pkg/util/awsutil/s3factory"
+	"github.com/coreos/etcd-operator/pkg/util/etcdutil"
+	"github.com/coreos/etcd-operator/pkg/util/k8sutil"
 
 	"k8s.io/client-go/kubernetes"
 )
 
 // TODO: replace this with generic backend interface for other options (PV, Azure)
 // handleS3 backups up etcd cluster to s3 and return s3 path for the backup file.
-func handleS3(kubecli kubernetes.Interface, s3 *api.S3Source, namespace, clusterName string) (string, error) {
+func handleS3(kubecli kubernetes.Interface, s3 *api.S3Source, clientTLSSecret, namespace, clusterName string) (string, error) {
 	cli, err := s3factory.NewClientFromSecret(kubecli, namespace, s3.AWSSecret)
 	if err != nil {
 		return "", err
 	}
 	defer cli.Close()
-	// TODO: support TLS.
-	bm := backup.NewBackupManagerFromWriter(kubecli, writer.NewS3Writer(cli.S3), clusterName, namespace)
+
+	var tlsConfig *tls.Config
+	if len(clientTLSSecret) != 0 {
+		d, err := k8sutil.GetTLSDataFromSecret(kubecli, namespace, clientTLSSecret)
```

Comment:
It is a previous mistake. But do you mind adding docs for GetTLSDataFromSecret() here? I feel this func is unclear. Thanks :)

Reply:
I don't quite understand. do you want me to doc string `GetTLSDataFromSecret`? or adding comments on here explaining what is going on?

Reply:
>  doc string GetTLSDataFromSecret()

Yeah. This one.